model/src/w3gdatmd.F90

```diff
@@ -1566,88 +1501,59 @@ SUBROUTINE W3NMOD ( NUMBER, NDSE, NDST, NAUX )
     !/ End of W3NMOD ----------------------------------------------------- /
     !/
   END SUBROUTINE W3NMOD
-  !/ ------------------------------------------------------------------- /
+
+  !> Initialize an individual spatial grid at the proper dimensions.
+  !>
+  !> Allocate directly into the structure array GRIDS. Note that
+  !> this cannot be done through the pointer alias!
+  !>
+  !> @note
+  !> - Grid dimensions apre passed through parameter list and then
+  !>   locally stored to assure consistency between allocation and
+  !>   data in structure.
+  !> - W3SETG needs to be called after allocation to point to
+  !>   proper allocated arrays.
+  !>
+  !> Switches: !/S    Enable subroutine tracing.
+  !>
+  !> ## Subroutine History
+  !> Date | Modification | Version
+  !> -----|--------------|--------
+  !> 24-Jun-2005 | Origination.                        | 3.07
+  !> 18-Jul-2006 | Add input grids.                    | 3.10
+  !> 05-Oct-2006 | Add filter to array pointers.       | 3.10
+  !> 02-Feb-2007 | Add FLAGST.                         | 3.10
+  !> 30-Oct-2009 | Implement run-time grid selection. (W. E. Rogers & T. J. Campbell, NRL) | 3.14
+  !> 30-Oct-2009 | Implement curvilinear grid type. (W. E. Rogers & T. J. Campbell, NRL)    | 3.14
+  !> 30-Oct-2009 | Implement unstructured grids        | 3.14.1
+  !> 03-Sep-2012 | Clean up of UG grids                | 4.08
+  !> 10-Dec-2014 | Add checks for allocate status      | 5.04
+  !>
+  !> @param[in] imod Model number to point to.
+  !> @param[in] mx Like NX in data structure.
+  !> @param[in] my Like NY in data structure.
+  !> @param[in] msea Like NSEA in data structure.
+  !> @param[in] ndse Error output unit number.
+  !> @param[in] ndst Test output unit number.
+#ifdef W3_SMC
```

Comment:
Does having the `#ifdef W3_SMC` here affect the generation of the documentation? Obviously it is not defined when the docs are built...

Reply:
Excellent question!

The answer is, we can cause doxygen to build documentation with whatever ifdefs we want, regardless of how we are building the software. Or we can build it with the switches in use when the software is built.

My concern at the moment is just capturing the info. How the documentation is built (i.e. with what switches) is an open question at the moment...

Reply:
That's a great question.    If something is only defined for SMC grid, then maybe that hsould be captured in text not by having an if-def in the documentation.  This seems a bit odd to me and I'm not sure how this would effect documentation - maybe doxygen takes care of this? I'm not an expert for sure...

Reply:
Well in this case we need an ifdef, because the parameters are ifdefed.

The @param statements must exactly match the parameters in the subprogram. So if we ifdef out some parameters, we ifdef out some documentation.

In practice, I think you would find life a lot easier if you did not ifdef subrprogram arguments. Instead, have them always present, and just the use ifdefs to control whether they are used in the subprogram.

Reply:
@MatthewMasarik-NOAA  - You have done the vast majority of the documentation.  How have you handled this so far?   I don't remember seeing anything with switches or ifdefs.

Reply:
One related circumstance is when an `#ifdef <switch>` block falls across a subroutine headers' input arg list.  This incurs a doxygen warning I've found no way to resolve without materially changing the header.

Reply:
Maybe we need to just add a `[SMC switch only]` comment to each of those SMC parameters then?
It's not ideal that we have preprocessor directives that change the subroutine call signature, but that's an issue for another time!

Reply:
OK done.

Reply:
> @MatthewMasarik-NOAA - You have done the vast majority of the documentation. How have you handled this so far? I don't remember seeing anything with switches or ifdefs.

I don't have a good fix for all cases, and have tried not to let special cases slow my progress.